mypy/build.py

```diff
@@ -1321,8 +1321,10 @@ def validate_meta(meta: Optional[CacheMeta], id: str, path: Optional[str],
         st = manager.get_stat(path)
     except OSError:
         return None
-    if not (stat.S_ISREG(st.st_mode) or stat.S_ISDIR(st.st_mode)):
-        manager.log('Metadata abandoned for {}: file {} does not exist'.format(id, path))
+    if not stat.S_ISDIR(st.st_mode) and not stat.S_ISREG(st.st_mode):
+        manager.log(
+            'Metadata abandoned for {}: file or directory {} does not exist'.format(id, path)
```

Comment:
this seems unrelated? Useful - but not related.

Reply:
I think there was a change here that was needed, but then got merged in another PR

Reply:
Yeah - I think remove this from this commit/PR before landing.
Better to keep the diff scoped to what's necessary.